scripts/prettier.js

```diff
@@ -55,4 +28,67 @@ async function runPrettierForAll() {
   });
 }
 
-runPrettierForAll();
+function parseArgs() {
+  return require('yargs')
+    .usage('Usage: prettier [commitHash] [options]')
+    .example('prettier', 'run prettier only on changed files')
+    .example('prettier HEAD~3', 'run prettier only on changed files since HEAD~3')
+    .options({
+      all: {
+        description: 'Run prettier on all projects',
+        boolean: true,
+      },
+    })
+    .alias('h', 'help').argv;
+}
+
+/**
+ *
+ * @param {PQueue} queue
+ */
+async function runOnChanged(queue) {
+  const prettierIntroductionCommit = 'HEAD~1';
+  const passedDiffTarget = parsedArgs._.length ? parsedArgs._[0] : prettierIntroductionCommit;
+
+  const projectRootPath = path.resolve(__dirname, '..');
```

Comment:
This ought to use `findGitRoot` from `./monorepo` so it's less sensitive to the script file being moved.

Reply:
just a heads up, I didn't change anything from how it was executed - copy pasted into more contextual functions. I know my bad :) , I should have only added those arg parsing additions.